modules/environment/aws/code-services/main.tf

```diff
@@ -202,3 +202,52 @@ resource "aws_sqs_queue_policy" "code_services_queue_policy" {
 }
 POLICY
 }
+
+resource "aws_iam_role" "event_mapper_role" {
+  count  = var.enable_aws_code_services ? 1 : 0
+  name = "${var.cluster}_event_mapper_role"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "arn:aws:iam::${var.account_id}:oidc-provider/${replace(var.openid_connect_provider_url, "https://", "")}"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "${replace(var.openid_connect_provider_url, "https://", "")}:aud": "sts.amazonaws.com"
```

Comment:
while I think it's fine to validate the audience, we should absolutely also be validating the subject (`sub`) claim to ensure that only the aws event mapper service account gets these permissions.  take a look at how the other `aws_iam_role` resources within the aws stack are doing this.

Reply:
I updated the trust policy to validate the subject claim